lib/serverengine/daemon_logger.rb

```diff
@@ -58,6 +64,16 @@ def logdev=(logdev)
         old_file_dev.close if old_file_dev
         @logdev = @file_dev
       end
+      if ServerEngine::linux?
+        if logdev.is_a?(String)
+          @inotify.close if @inotify
+          @inotify = INotify::Notifier.new
+          @inotify.watch(logdev, :move_self) do |event|
+            close
+            reopen
+          end
+        end
+      end
```

Comment:
Can we extract as a method here?
Current implementation is a bit of messy for me.